lnwallet/size.go

```diff
@@ -0,0 +1,107 @@
+package lnwallet
+
+import (
+	"github.com/roasbeef/btcd/blockchain"
+)
+
+const (
+	WitnessFactor      = blockchain.WitnessScaleFactor
+	MaxTransactionCost = blockchain.MaxBlockCost / 10
+
+	// The weight(cost), which is different from the !size! (see BIP-141),
+	// is calculated as:
+	// Weight = 4 * BaseSize + WitnessSize (weight).
```

Comment:
Actually, the two are _nearly_ interchangeable. The method as described in this comment may fall short of the actual weight value as it doesn't factor in the bytes within the serialized transaction form required to signal the arity of the witness stack and the size of each element.